pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/core.go

```diff
@@ -42,18 +47,49 @@ func StartControllers(scCfg *rest.Config, client kubernetes.Interface,
 
 	klog.V(2).Info("Routable pod controllers start with VPC mode enabled: ", vpcModeEnabled)
 
-	ippManager, err := ippoolmanager.GetIPPoolManager(vpcModeEnabled, scCfg, clusterNS, podIPPoolType)
-	if err != nil {
-		return fmt.Errorf("fail to get ippool manager or start ippool controller: %w", err)
-	}
+	ctx := informerManager.GetContext()
+	var nsxIPManager nsxipmanager.NSXIPManager
+	if vpcModeEnabled {
```

Comment:
Could we extract following `if` `else` block to two small functions:

```go
func startIPAddressAllocationController()

func startIPPoolController()
```

to improve the code readability?